Dev

fangzhouli opened the pull request:
Hi Jason, this is the pull request specifically regarding the command line interface (I am still finalizing the refactoring for the back end, so you can skip them!) All command-line codes are at app/client/cli.

Since this is a new branch, to address the issue on the last code review of dev-insertion:
1. This is another huge file structure changing, so I will make sure to give you all specific file path regarding each addressed issue.
2. PEP8 linting: All resolved, however, I am still sticking to "--max-line-length == 80".
3. Global constants: in cli/models/converter.py, all capitalized.
4. Suffix underline in data template headers: Since last time we discussed and chose a new data template, no suffix needed anymore.
5. Naming: se -> df_compound (see changed file for more info.)
6. JSON file suffix: No JSON file will be generated anymore. The CSV file will be read and converted, and the converted is directly sent to the back end.
7. Logging: Added logging in each Python file, but please give me additional advice since I am not sure if I am using correctly.
8. Newline at the end of each file: Resolved in CLI, will changed in JS files.

Best,
Fang

jasonyoun committed:
code review added

fangzhouli committed:
ignored node files

fangzhouli committed:
reorganized file structure, added insertion functionality

fangzhouli committed:
removed garbage files

fangzhouli committed:
improved style and debugged upload.py

fangzhouli committed:
implemented create function

fangzhouli committed:
renamed upload.py to converter, modularized and added convert_csv_to_json

fangzhouli committed:
added get_headers.py

fangzhouli committed:
for path configuration

fangzhouli committed:
removed class, changed to functional programming

fangzhouli committed:
added cli for gen and create

fangzhouli committed:
testing

fangzhouli committed:
moved header information into models

fangzhouli committed:
added TODO and return schema

fangzhouli committed:
added TODO and return schema

fangzhouli committed:
added todo

fangzhouli committed:
added misc files

fangzhouli committed:
added comments and moved header to controllers

fangzhouli committed:
added trivial changes

fangzhouli committed:
new readme for internal use

fangzhouli committed:
added minor changes

fangzhouli committed:
changed to absolute path

jasonyoun commented:
I tried to follow the README file to reproduce the results and ran into two separate error.

1. Running the server locally using `node controllers.js`.
```
module.js:549
    throw err;
    ^

Error: Cannot find module 'express'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/jyoun/Jason/Research/TIP/tip/app/server/controllers.js:6:17)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
```
2. Running the `tip-cli`.
```
Traceback (most recent call last):
  File "/home/jyoun/.local/bin/tip-cli", line 33, in <module>
    sys.exit(load_entry_point('tip-client-cli==0.0.1', 'console_scripts', 'tip-cli')())
  File "/home/jyoun/.local/bin/tip-cli", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/home/jyoun/.local/lib/python3.6/site-packages/importlib_metadata/__init__.py", line 95, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 941, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'views'
```

Other comments I had were,,,

1. Is the Python requirement >= 3.8.1  absolutely necessary? I am testing it on my local machine which is running 3.6.9. We should try to include at least 3.6. since 3.8 is quite recent (October, 2019).

2. How do I uninstall tip-cli? It'll be good to include the uninstall instructions in the README file.

Please address my comments.

Thanks!

jasonyoun commented:
As for the first error which says I am missing the `express` module, am I expected to install it myself? I can make it work, but strictly following the README, people will run into the same problem. You can install these module locally right? I don't know all the details, so please let me know if I am mistaken.

fangzhouli committed:
updated usage, added uninstallation instruction

fangzhouli committed:
added __init__ to enable bin

fangzhouli committed:
allowed py3.6

fangzhouli commented:
> I tried to follow the README file to reproduce the results and ran into two separate error.
> 
> 1. Running the server locally using `node controllers.js`.
> 
> ```
> module.js:549
>     throw err;
>     ^
> 
> Error: Cannot find module 'express'
>     at Function.Module._resolveFilename (module.js:547:15)
>     at Function.Module._load (module.js:474:25)
>     at Module.require (module.js:596:17)
>     at require (internal/module.js:11:18)
>     at Object.<anonymous> (/home/jyoun/Jason/Research/TIP/tip/app/server/controllers.js:6:17)
>     at Module._compile (module.js:652:30)
>     at Object.Module._extensions..js (module.js:663:10)
>     at Module.load (module.js:565:32)
>     at tryModuleLoad (module.js:505:12)
>     at Function.Module._load (module.js:497:3)
> ```

Issue addressed by adding `npm install` in README.

> 1. Running the `tip-cli`.
> 
> ```
> Traceback (most recent call last):
>   File "/home/jyoun/.local/bin/tip-cli", line 33, in <module>
>     sys.exit(load_entry_point('tip-client-cli==0.0.1', 'console_scripts', 'tip-cli')())
>   File "/home/jyoun/.local/bin/tip-cli", line 25, in importlib_load_entry_point
>     return next(matches).load()
>   File "/home/jyoun/.local/lib/python3.6/site-packages/importlib_metadata/__init__.py", line 95, in load
>     module = import_module(match.group('module'))
>   File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 941, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>   File "<frozen importlib._bootstrap>", line 994, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 971, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
> ModuleNotFoundError: No module named 'views'
> ```

Issue addressed by adding __init__.py in views.

> Other comments I had were,,,
> 
> 1. Is the Python requirement >= 3.8.1  absolutely necessary? I am testing it on my local machine which is running 3.6.9. We should try to include at least 3.6. since 3.8 is quite recent (October, 2019).

Changed to 3.6.

> 2. How do I uninstall tip-cli? It'll be good to include the uninstall instructions in the README file.

Added uninstallation.

jasonyoun committed:
Merge branch 'master' into dev
